From 45a3afa1100df0c072b9df326868c24d1fab7da1 Mon Sep 17 00:00:00 2001
From: Patrice Chotard <patrice.chotard@foss.st.com>
Date: Thu, 18 Jan 2024 09:29:26 +0100
Subject: [PATCH 0793/1141] mtd: hyperbus: hyperbus-sfdp: Fix
 hyperbus_s26hx_chip_setup()

n256ks variable is defined as u8, this is not enough in case
of HyperFlash with more than 255 256KB sectors.

Update n256ks variable to u16 to be able to detect S26HSL512
HyperFlash with 256 256KB sectors.

Change-Id: Ib8bfed3bb04f973e35bbcfb6d8c74a2b730f5ac4
Signed-off-by: Patrice Chotard <patrice.chotard@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/353481
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/mtd/hyperbus/hyperbus-sfdp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/hyperbus/hyperbus-sfdp.c b/drivers/mtd/hyperbus/hyperbus-sfdp.c
index 5ba010aeeb3c..7efcc3c2cc0f 100644
--- a/drivers/mtd/hyperbus/hyperbus-sfdp.c
+++ b/drivers/mtd/hyperbus/hyperbus-sfdp.c
@@ -102,7 +102,8 @@ static int __xipram hyperbus_s26hx_chip_setup(struct map_info *map,
 					      struct cfi_private *cfi)
 {
 	u16 mfr_id, dev_id1, dev_id2, cfr1v, cfr2v;
-	u8 nregions, nbtm4ks, ntop4ks, n256ks;
+	u8 nregions, nbtm4ks, ntop4ks;
+	u16 n256ks;
 	u32 *erase;
 
 	/* Read manufacturer and Device ID */
-- 
2.39.2

